tensorflow_datasets/structured/movielens_parsing_test.py

```diff
@@ -0,0 +1,507 @@
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for movielens_parsing."""
+
+import os
+
+import tensorflow_datasets.public_api as tfds
+from tensorflow_datasets.structured import movielens_parsing
+
+class MovieLensUtilsTest(tfds.testing.TestCase):
+  """Tests for helper functions in movielens_parsing"""
+
+  def test_parse_current_movies_data(self):
+    """Test for parse_current_movies_data"""
+    latest_small_path = os.path.join(
+        tfds.testing.fake_examples_dir(),
+        'movie_lens',
+        'ml-latest-small',
+    )
+    movies_generator = movielens_parsing.parse_current_movies_data(
+        latest_small_path
+    )
+    expected_result = [
+        (0, {
+            'movie_id': '1',
+            'movie_genres': ['Fantasy', 'War'],
+            'movie_title': 'Horse Movie (2011)',
+        }),
+        (1, {
+            'movie_id': '2',
+            'movie_genres': ['Adventure', 'Documentary'],
+            'movie_title': 'Eagle Movie (1999)',
+        }),
+        (2, {
+            'movie_id': '3',
+            'movie_genres': ['(no genres listed)'],
+            'movie_title': 'Test Movie (2014)',
+        }),
+        (3, {
+            'movie_id': '4',
+            'movie_genres': ['Fantasy', 'Horror', 'Musical'],
+            'movie_title': 'Fake Movie (2007)',
+        }),
+        (4, {
+            'movie_id': '5',
+            'movie_genres': ['Mystery'],
+            'movie_title': 'Squirrel Movie (2017)',
+        }),
+        (5, {
+            'movie_id': '6',
+            'movie_genres': ['Drama', 'War'],
+            'movie_title': 'Test Movie (2003)',
+        }),
+        (6, {
+            'movie_id': '7',
+            'movie_genres': ['Horror'],
+            'movie_title': 'Rabbit Movie (1999)',
+        }),
+        (7, {
+            'movie_id': '8',
+            'movie_genres': ['Mystery', 'War'],
+            'movie_title': 'Tree Movie (2001)',
+        }),
+    ]
+    parsed_example_list = list(movies_generator)
+    self.assertListEqual(expected_result, parsed_example_list)
```

Comment:
FYI `self.assertEqual` would work here (more details in the [python docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertListEqual))

Reply:
I see. I changed it to assertEqual to be concise and to be compatible to potential future changes.